BugFix: prevent flash on tab switch in multi-view

SlySven opened the pull request:
If we are showing all the profiles at the same time then hiding the current one does not make sense when in multi-view mode...!

This will close #4792.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

#### Release post highlight

Remove flash on profile switching in multi-view mode.

SlySven committed:
BugFix: prevent flash on tab switch in multi-view

If we are showing all the profiles at the same time then hiding the current
one does not make sense when in multi-view mode...!

This will close #4792.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven committed:
Revise: remove else-after-return introduced in previous commit

Keeps the GHA clang-tidy bot happy.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

SlySven commented on src/mudlet.cpp:
```diff
@@ -1343,6 +1343,9 @@ void mudlet::slot_tab_changed(int tabID)
     if (!pHost || !pHost->mpConsole) {
         mpCurrentActiveHost = nullptr;
         return;
+    } else if (mpCurrentActiveHost && (mpCurrentActiveHost.data() == pHost)) {
```
Done.

vadi2 reviewed (commented):
It works amazing!

I understand the comments were useful there for debugging - could you take them out now?

SlySven commented:
I'll take a look and prune some of them - but there is one or two that ought to stay IMHO.

SlySven committed:
Revise: try and reduce comments in PR

I'd like to leave some of them in place because there is some aspects that
are not blindingly obvious to the casual view. Also there is some code who
presence is not entirely clear to be needed to me - until someone can work
if it has to be there then it is probably okay to leave notes about what I
have worked out around it for whoever wants to do the further
investigation.

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

vadi2 commented on src/mudlet.cpp:
```diff
@@ -1382,7 +1378,7 @@ void mudlet::slot_tab_changed(int tabID)
     }
 
-    // This *seems* to be redundent:
+    // CHECK: This *seems* to be redundent - further investigation needed to be sure:
```
```suggestion
    // CHECK: This *seems* to be redundant - further investigation needed to be sure:
```

SlySven replied:
:man_facepalming:

SlySven committed:
Revise: fix typo in comment

Signed-off-by: Stephen Lyons <slysven@virginmedia.com>

Co-authored-by: Vadim Peretokin <vperetokin@gmail.com>
